include/canbus/isotp_fast.h

```diff
@@ -74,8 +74,8 @@ typedef void (*isotp_fast_send_callback_t)(int result, void *arg);
  * @param ctx Pointer to the main context.
  * @param addr Address.
  */
-typedef struct isotp_fast_addr (*isotp_fast_get_tx_addr_callback_t)(struct isotp_fast_ctx *ctx,
-                                                                    struct isotp_fast_addr addr);
+typedef struct isotp_fast_addr (*isotp_fast_get_tx_addr_callback_t)(
+    const struct isotp_fast_addr *addr);
```

Comment:
What is the rationale for making this a pointer? (for my interest/education rather than any particular critique)

Reply:
Efficiency. As I recently learned from some Zephyr folks, C compilers usually don't support [return value optimization / copy elision](https://en.wikipedia.org/wiki/Copy_elision). This means that if a struct doesn't fit into one MCU register we need more expensive copy operations when returning structs from functions or when passing structs to functions by value. In our case this would *probably* only happen when extended addressing is enabled, as in that case the struct would be larger than 32 bit. If we use pointers, we are sure we don't have any unnecessary copy operations.

Ideally we'd even use a pointer for the return struct, but I kept it as is to make the code easier to read.